context-config-store/store/context_config_store_impl.go

```diff
@@ -0,0 +1,139 @@
+	return &contextConfigStoreImpl{
+		RWMutex: &sync.RWMutex{},
+		storage: storage,
+	}
+}
+
+func (store *contextConfigStoreImpl) GetAllContexts() ([]*api.KurtosisContext, error) {
+	store.RLock()
+	defer store.RUnlock()
+
+	contextConfig, err := store.storage.LoadContextConfig()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Unable to load current context config")
+	}
+	return contextConfig.GetContexts(), nil
+}
+
+func (store *contextConfigStoreImpl) GetCurrentContext() (*api.KurtosisContext, error) {
+	store.RLock()
+	defer store.RUnlock()
+
+	contextConfig, err := store.storage.LoadContextConfig()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Unable to load current context config")
+	}
+
+	currentContextUuid := contextConfig.GetCurrentContext()
+	var knownContexts []string
+	for _, kurtosisContext := range contextConfig.GetContexts() {
+		knownContexts = append(knownContexts, kurtosisContext.GetUuid().GetValue())
+		if kurtosisContext.GetUuid().GetValue() == currentContextUuid.GetValue() {
+			return kurtosisContext, nil
+		}
+	}
+	return nil, stacktrace.NewError("Unable to find current context info in context config file. "+
+		"Current context is set to '%s' but known contexts are: '%s'",
+		currentContextUuid.GetValue(),
+		strings.Join(knownContexts, ", "))
+}
+
+func (store *contextConfigStoreImpl) SwitchContext(contextUuid *api.ContextUuid) error {
+	store.Lock()
+	defer store.Unlock()
+
+	contextConfig, err := store.storage.LoadContextConfig()
+	if err != nil {
+		return stacktrace.Propagate(err, "Unable to load current context config")
+	}
+
+	var doesExist bool
+	for _, kurtosisContext := range contextConfig.GetContexts() {
+		doesExist = doesExist || kurtosisContext.GetUuid().GetValue() == contextUuid.Value
+	}
+	if !doesExist {
+		return stacktrace.NewError("Context with UUID '%s' does not exist yet.", contextUuid.GetValue())
+	}
+
+	newContextConfig := api.NewKurtosisContextConfig(contextUuid, contextConfig.GetContexts()...)
+	if err = store.storage.PersistContextConfig(newContextConfig); err != nil {
+		return stacktrace.Propagate(err, "Unable to persist new current context to store")
+	}
+	return nil
+}
+
+func (store *contextConfigStoreImpl) AddNewContext(contextToAdd *api.KurtosisContext) error {
+	store.Lock()
+	defer store.Unlock()
+
+	contextConfig, err := store.storage.LoadContextConfig()
+	if err != nil {
+		return stacktrace.Propagate(err, "Unable to load current context config")
+	}
+
+	newContextUuid := contextToAdd.GetUuid()
+	for _, kurtosisContext := range contextConfig.GetContexts() {
+		if kurtosisContext.GetUuid().GetValue() == newContextUuid.GetValue() {
+			return nil // doing nothing as context already exists
+		}
+	}
+
+	var newContexts []*api.KurtosisContext
+	for _, kurtosisContext := range contextConfig.GetContexts() {
+		newContexts = append(newContexts, kurtosisContext)
+	}
+	newContexts = append(newContexts, contextToAdd)
```

Comment:
are all of these the new context or is only the contextToAdd the one?

Reply:
contextToAdd is the only new one. I changed the variable name to be `updatedContextsList`, hopefully it helps